docs/sphinx/source/contributing/style_guide.rst

```diff
@@ -70,6 +70,26 @@ the ``docs/readthedocs.org:pvlib-python`` link within the checks
 status box at the bottom of the pull request.
 
 
+.. _documentation-units:
+
+Parameter names and units
+-------------------------
+
+When specifying parameters and their units, please follow these guidelines:
+
+- Use the recommended parameter name and units by :ref:`variables_style_rules` where possible.
+- Enclose units in square brackets, e.g. ``[W]``.
+- Use unicode superscripts symbols for exponents, e.g. ``m²``.
+
+  These superscripts characters are ``⁰``, ``¹``, ``²``, ``³``, ``⁴``, ``⁵``, ``⁶``, ``⁷``, ``⁸``, ``⁹``, ``⁺``, ``⁻``. Degree symbol is ``°``.
+
+- Link to a brief description of the magnitude in the :ref:`variables_style_rules` section if it exists, via the sphinx role ``:term:`glossary_term```. For example, to document ``dni`` use:
```

Comment:
It seems that this PR is dependent on #2234 being merged first.
Currently, the `:term:` example would not work with the linked variables and style rules page. I guess you know this(?) but I thought I'd highlight this just in case. The link would need to be updated to link to the glossary (or whatever we end up calling it).
I guess the alternative would be to omit this point for now, then add it in later as a follow up

Reply:
It is dependant, that's true. I'll change to draft until it gets into main to avoid making too many changes that belong to the same thing. Thanks in fact for highlighting it, when I wrote it I didn't distinguish between the two pages.